torch/distributed/_composable/fsdp/_fsdp_param.py

```diff
@@ -266,12 +284,37 @@ def init_all_gather_outputs(
     @torch.no_grad()
     def init_unsharded_param(self):
         if hasattr(self, "_unsharded_param"):
-            return  # already initialized
-        # For the default path (no post-all-gather), the all-gather output
-        # gives the unsharded parameter data directly
-        assert len(self.all_gather_outputs) == 1
+            if not self._use_all_gather_extensions:
+                return  # already initialized
```

Comment:
Some things like this can be cleaned up. E.g., for `not self._use_all_gather_extensions`, `len(self._unsharded_inner_tensors) == 0`, so the code below is a no-op.